lib/parser/lexer.rl

```diff
@@ -87,6 +87,11 @@ class Parser::Lexer
     'v' => "\v", '\\' => "\\"
   }
 
+  BLANK_STRING = ''.freeze
+  ESCAPED_NEXT_LINE = "\\\n".freeze
+  REGEXP_META_CHARACTERS = "\\$()*+.<>?[]^{|}".freeze
```

Comment:
Can you look at replacing `REGEXP_META_CHARACTERS.include?` with a regular expression match? It might be faster to do that.